data/src/test/java/io/homeassistant/companion/android/data/integration/IntegrationRepositoryImplSpec.kt

```diff
@@ -0,0 +1,87 @@
+package io.homeassistant.companion.android.data.integration
+
+import io.homeassistant.companion.android.data.LocalStorage
+import io.homeassistant.companion.android.domain.authentication.AuthenticationRepository
+import io.homeassistant.companion.android.domain.integration.DeviceRegistration
+import io.mockk.coEvery
+import io.mockk.coVerifyAll
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+object IntegrationRepositoryImplSpec : Spek({
+
+    describe("a repository") {
+
+        val localStorage by memoized { mockk<LocalStorage>(relaxUnitFun = true) }
+        val integrationService by memoized { mockk<IntegrationService>(relaxUnitFun = true) }
+        val authenticationRepository by memoized { mockk<AuthenticationRepository>(relaxUnitFun = true) }
+
+        val repository by memoized {
+            IntegrationRepositoryImpl(
+                integrationService,
+                authenticationRepository,
+                localStorage
+            )
+        }
+
+        describe("registerDevice") {
+            beforeEachTest {
+                val deviceRegistration = DeviceRegistration(
+                    "appId",
+                    "appName",
+                    "appVersion",
+                    "deviceName",
+                    "manufacturer",
+                    "model",
+                    "osName",
+                    "osVersion",
+                    false,
+                    null
+                )
+                coEvery {
+                    integrationService.registerDevice("ABC123", any())
```

Comment:
You should check the binding between DeviceRegistration and RegisterDeviceRequest.

Reply:
Done.